packages/replay/src/metadata/source.ts

```diff
@@ -22,31 +22,49 @@ class GitHubHttpError extends Error {
   }
 }
 
-// Add a basic cache so we don't refetch data from GH repeatedly for the same resources
-const gFetchCache: Record<string, { json: any | null; status: number; statusText: string }> = {};
-async function fetchWithCache(
+type CacheEntry = { json: any | null; status: number; statusText: string };
+
+export const cache: Map<string, CacheEntry> = new Map();
+
+// TODO [PRO-676] Import this from the "shared" package
+export async function fetchWithCacheAndRetry(
   url: string,
-  init?: RequestInit
-): Promise<{ json: any | null; status: number; statusText: string }> {
-  if (!(url in gFetchCache)) {
+  init?: RequestInit,
+  options: {
+    baseDelay?: number;
+    maxAttempts?: number;
+  } = {}
+): Promise<CacheEntry> {
+  const { baseDelay = 1_000, maxAttempts = 3 } = options;
+
+  let attempt = 0;
+
+  while (!cache.has(url)) {
+    attempt++;
+
     const resp = await fetch(url, init);
     if (resp.status === 200) {
       const json = await resp.json();
-      gFetchCache[url] = {
+      cache.set(url, {
+        json,
         status: resp.status,
         statusText: resp.statusText,
-        json,
-      };
+      });
+    } else if (attempt < maxAttempts) {
+      // Retry with exponential backoff (e.g. 1s, 2s, 4s, ...)
+      const delay = Math.pow(2, attempt) * baseDelay;
+      await new Promise(resolve => setTimeout(resolve, delay));
     } else {
-      gFetchCache[url] = {
+      // If we've run out of retries, store and return the error
+      cache.set(url, {
         json: null,
         status: resp.status,
         statusText: resp.statusText,
-      };
+      });
```

Comment:
The old function cached so I assumed that was behavior we wanted. I don't feel strongly about that though.

Reply:
I see, yeah, no strong objection either because the same request would likely fail again. 

Also because we use URL as cache key and when we move this function to a shared package, it'll be easy to make mistakes re-using this same function with like GraphQL, we have the same URL for all queries/mutations. But it'll be evident immediately. I'll defer to you.

Reply:
We would not want to use this utility method for GraphQL queries since that would likely cause caching problems even for successful responses.

Reply:
PR response:
* Documented that the method should not be used with GraphQL
* Added a method that enables all cached URLs, or an individual URL, to be evicted.